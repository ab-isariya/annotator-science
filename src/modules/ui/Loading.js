import PropTypes from 'prop-types';

const Loading = ({className}) => <div className={className}>Loading...</div>;

Loading.propTypes = {
  //TailwindCSS classnames
  className: PropTypes.string
};

export default Loading;
