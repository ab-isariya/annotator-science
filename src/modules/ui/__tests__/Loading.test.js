import Loading from '../Loading';

import renderer from 'react-test-renderer';

describe('<Loading/>', () => {
  it('Renders without crashing', () => {
    const comp = renderer.create(<Loading />).toJSON();

    expect(comp).toMatchSnapshot();
  });

  it('Renders with a custom classname', () => {
    const comp = renderer.create(<Loading className="testClass" />).toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.props.className).toMatch(/testClass/i);
  });
});
