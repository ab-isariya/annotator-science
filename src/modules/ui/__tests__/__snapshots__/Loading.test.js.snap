// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading/> Renders with a custom classname 1`] = `
<div
  className="testClass"
>
  Loading...
</div>
`;

exports[`<Loading/> Renders without crashing 1`] = `
<div>
  Loading...
</div>
`;
